sourced/ml/transformers/moder.py

```diff
@@ -73,21 +74,28 @@ def extract_functions_from_row(self, row: Row):
         template = row.asDict()
         for func, name in self.extract_functions_from_uast(uast):
             data = template.copy()
-            data[EngineConstants.Columns.Uast] = [self.serialize_uast(func)]
+            data[EngineConstants.Columns.Uast] = [bytearray(self.serialize_uast(func))]
             data[EngineConstants.Columns.BlobId] += "_%s:%d" % (name, func.start_position.line)
             yield Row(**data)
 
-    def extract_functions_from_uast(self, uast: "bblfsh.Node"):
-        allfuncs = list(self.filter_uast(uast, self.FUNC_XPATH))
+    @staticmethod
+    def extract_functions_from_uast(uast: "bblfsh.Node"):
+        node_seq = uast2sequence(uast)
+        allfuncs = [node for node in node_seq if FUNCTION in node.roles and
+                    DECLARATION in node.roles]
         internal = set()
         for func in allfuncs:
             if id(func) in internal:
                 continue
-            for sub in self.filter_uast(func, self.FUNC_XPATH):
```

Comment:
Can you please add a flag to switch between xpath and hotfix implementations? E.g.

```
class Moder:
    USE_XPATH = os.getenv("USE_XPATH", False) in ("1", "true", "yes")
```

Reply:
Done